src/plugins/vis_type_vega/public/vega_view/vega_map_view.js

```diff
@@ -21,8 +21,34 @@ import { i18n } from '@kbn/i18n';
 import { vega } from '../lib/vega';
 import { VegaBaseView } from './vega_base_view';
 import { VegaMapLayer } from './vega_map_layer';
-import { getEmsTileLayerId, getUISettings } from '../services';
-import { lazyLoadMapsLegacyModules } from '../../../maps_legacy/public';
+import { getMapsLegacyConfig, getUISettings } from '../services';
+import { lazyLoadMapsLegacyModules, TMS_IN_YML_ID } from '../../../maps_legacy/public';
+
+const isUserConfiguredTmsLayer = ({ tilemap }) => Boolean(tilemap.url);
+
+const getMapStyleOptions = async (serviceSettings, mapConfig, isDarkMode) => {
+  const mapsLegacyConfig = getMapsLegacyConfig();
+  const tmsServices = await serviceSettings.getTMSServices();
+
+  let mapStyle;
+
+  if (mapConfig.mapStyle !== 'default') {
```

Comment:
**to discuss:** in accordance with the following documentation https://www.elastic.co/guide/en/kibana/6.8/vega-with-a-map.html `mapStyle` param accepts just a  2 values: `false` and `default`. 

It looks a little odd to me because in some cases it might be useful to let the user set a specific tms layer value.  For example, if I set `map.tilemap.url` param and want to create some vis with that layer and some with the default one (`road_map`) 

```
{
  $schema: https://vega.github.io/schema/vega/v5.json
  config: {
    kibana: {
      type: map
      mapStyle: road_map
    }
  }
}
```

To add support of that we just need to remove the following `if` statement from `vis_type_vega/public/data_model/vega_parser.ts`

  ```
  if (res.mapStyle !== `default` && res.mapStyle !== false) {
      this._onWarning(
        i18n.translate('visTypeVega.vegaParser.mapStyleValueTypeWarningMessage', {
          defaultMessage:
            '{mapStyleConfigName} may either be {mapStyleConfigFirstAllowedValue} or {mapStyleConfigSecondAllowedValue}',
          values: {
            mapStyleConfigName: 'config.kibana.mapStyle',
            mapStyleConfigFirstAllowedValue: 'false',
            mapStyleConfigSecondAllowedValue: '"default"',
          },
        })
      );
      res.mapStyle = `default`;
    }
```

Reply:
I like the idea of allowing users to specify which base layer to use. I think this would be best completed in a separate PR so the implementation can be discussed outside of the context of this fix.

Reply:
agree, it will be done out of that PR, just wanted to discuss it firstly